sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/Config/ServiceHubContextStore.cs

```diff
@@ -70,5 +70,34 @@ private async Task<IServiceHubContext> GetFromLazyAsync(string hubName, (Lazy<Ta
                 throw;
             }
         }
+
+        public async ValueTask DisposeAsync()
+        {
+            _serviceManager.Dispose();
+            foreach (var tuple in _store.Values)
+            {
+                if (tuple.Value is not null)
+                {
+                    await tuple.Value.DisposeAsync().ConfigureAwait(false);
+                }
+                if (tuple.Lazy is not null && tuple.Lazy.IsValueCreated)
+                {
+                    await (await tuple.Lazy.Value.ConfigureAwait(false)).DisposeAsync().ConfigureAwait(false);
+                }
+            }
+            foreach (var lazy in _stronglyTypedStore.Values)
+            {
+                if (lazy.IsValueCreated)
+                {
+                    // The IAsyncDisposable interface doesn't apply to ServiceHubContext<T> on netstandard2.0 yet.
+                    ((IDisposable)(await lazy.Value.ConfigureAwait(false))).Dispose();
+                }
+            }
+        }
+
+        public void Dispose()
+        {
+            DisposeAsync().AsTask().Wait();
```

Comment:
I think GetAwaiter().GetResult() is preferred to Wait.

right @pakrym?

Reply:
When `GetAwaiter().GetResult()` was used, I got an error : "Severity	Code	Description	Project	File	Line	Suppression State
Error	AZC0102	Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.	". But I found that `EnsureCompleted()` will throw an error if the task is not completed, which is not expected.

Reply:
In this case, you should be able to suppress the warning for AZC0102. The point of this warning is to recommend EnsureCompleted since it has a safeguard to not use for sync over async code. Here, we are intentionally doing sync over async, as there is no synchronous alternative for the methods we are calling.